core/templates/components/entity-creation-services/story-creation.service.spec.ts

```diff
@@ -68,62 +75,69 @@ describe('Story Creation Service', () => {
       .and.returnValue($q.resolve('sample-csrf-token'));
   }));
 
-  it('should not initiate new story creation if another is in process', () => {
-    spyOn($uibModal, 'open').and.returnValue({
-      result: $q.resolve({
-        isValid: () => true,
-        title: 'Title',
-        description: 'Description',
-        urlFragment: 'url'
-      })
-    });
+  it('should not initiate new story creation if another is in process',
+    fakeAsync(() => {
+      spyOn(ngbModal, 'open').and.returnValue(
+      {
+        result: Promise.resolve({
+          isValid: () => true,
+          title: 'Title',
+          description: 'Description',
+          urlFragment: 'url'
+        })
+      } as NgbModalRef
+      );
 
-    StoryCreationService.createNewCanonicalStory();
-    $scope.$apply();
+      StoryCreationService.createNewCanonicalStory();
+      tick();
 
-    // Creating a new story while previous was in creation process.
-    expect(StoryCreationService.createNewCanonicalStory()).toBe(undefined);
-  });
+      // Creating a new story while previous was in creation process.
+      expect(StoryCreationService.createNewCanonicalStory()).toBe(undefined);
+    })
+  );
 
   it('should post story data to server and change window location' +
-    ' on success', () => {
-    spyOn($uibModal, 'open').and.returnValue({
-      result: $q.resolve({
-        isValid: () => true,
-        title: 'Title',
-        description: 'Description',
-        urlFragment: 'url'
-      })
-    });
+    ' on success', fakeAsync(() => {
+    spyOn(ngbModal, 'open').and.returnValue(
+      {
+        result: Promise.resolve({
+          isValid: () => true,
+          title: 'Title',
+          description: 'Description',
+          urlFragment: 'url'
+        })
+      } as NgbModalRef
+    );
 
     $httpBackend.expectPOST('/topic_editor_story_handler/id')
       .respond(200, {storyId: 'id'});
 
     expect(mockWindow.location).toBe('');
 
-
     StoryCreationService.createNewCanonicalStory();
-    $scope.$apply();
-
-    $httpBackend.flush();
+    tick();
```

Comment:
I think we need `$scope.$apply();` after this.
(Since the service is still in angJS)

Reply:
It's still not working

Reply:
<pre>  it('should post story data to server and change window location' +
    ' on success', fakeAsync(() => {
    spyOn(ngbModal, 'open').and.returnValue(
      {
        result: Promise.resolve({
          isValid: () => true,
          title: 'Title',
          description: 'Description',
          urlFragment: 'url'
        })
      } as NgbModalRef
    );

    $httpBackend.expect('POST','/topic_editor_story_handler/id')
      .respond(200, {storyId: 'id'});

    expect(mockWindow.location).toBe('');

    StoryCreationService.createNewCanonicalStory();

    $scope.$apply()
    tick();
    $httpBackend.flush();

    expect(mockWindow.location).toBe('/story_editor/id');
  }));</pre>

Reply:
Done.